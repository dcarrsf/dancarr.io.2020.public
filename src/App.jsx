import { BrowserRouter as Router } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import { Provider } from 'react-redux';
import AppView from './app/components/AppView';
import ErrorBoundary from './app/components/Error';
import React from 'react';
import routes from './app/routes';
import store from './app/model/Store';
import styles from './styles/global';

const useStyles = makeStyles(styles);

export default function App() {
    useStyles();
    return (
        <ErrorBoundary>
            <Provider store={store}>
                <Router>
                    <AppView>{routes}</AppView>
                </Router>
            </Provider>
        </ErrorBoundary>
    );
}
