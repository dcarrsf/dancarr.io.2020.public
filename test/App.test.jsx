import { shallow } from 'enzyme';
import App from '../src/App';
import React from 'react';
import renderer from 'react-test-renderer';

describe('App', () => {
  test('matches its snapshot', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
	});
	
	describe('component', () => {
		let wrapper;
		beforeAll(() => {
				wrapper = shallow(<App />);
		});
		it('renders without error', () => {
			expect(wrapper).toHaveLength(1);
		});
	});
});
