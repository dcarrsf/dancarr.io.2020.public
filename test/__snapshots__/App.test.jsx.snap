// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches its snapshot 1`] = `
<main
  className="makeStyles-root-1"
>
  <header
    className="makeStyles-root-2"
  >
    <div
      className="makeStyles-root-3"
    >
      <a
        className="makeStyles-link-7"
        href="/"
        onClick={[Function]}
      >
        <h6
          className="MuiTypography-root makeStyles-name-8 makeStyles-selected-9 MuiTypography-h6"
        >
          HEADER
        </h6>
      </a>
      <div>
        <a
          className="makeStyles-link-4"
          href="/explore"
          onClick={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-5"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Explore
            </span>
          </button>
        </a>
        <a
          className="makeStyles-link-4"
          href="/preview"
          onClick={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-5"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              Preview
            </span>
          </button>
        </a>
      </div>
    </div>
  </header>
  <div
    className="makeStyles-root-10"
  >
    <div
      className="makeStyles-root-12"
    >
      <div
        className="makeStyles-content-13"
      >
        <div
          className="makeStyles-centerAlign-11"
        >
          <div>
            <h1
              className="MuiTypography-root MuiTypography-h1"
            >
              TEMPLATE
            </h1>
          </div>
        </div>
      </div>
      <footer
        className="makeStyles-root-14"
      >
        <span
          className="MuiTypography-root makeStyles-copyright-15 MuiTypography-caption"
        >
          Copyright Â© 2020 [template]. All rights reserved.
        </span>
      </footer>
    </div>
  </div>
</main>
`;
